tests/mr/device/mr_ref_test.hpp

```diff
@@ -416,4 +380,85 @@ inline auto make_binning()
+
+    mr = *owned_mr;
+  }
+
+  // Owned resource to use in tests, type determined by the type of factory function
+  std::invoke_result_t<MRFactoryFunc> owned_mr;
+};
+
+using cuda_mr        = rmm::mr::cuda_memory_resource;
+using pinned_mr      = rmm::mr::pinned_host_memory_resource;
+using cuda_async_mr  = rmm::mr::cuda_async_memory_resource;
+using managed_mr     = rmm::mr::managed_memory_resource;
+using pool_mr        = rmm::mr::pool_memory_resource<cuda_mr>;
+using pinned_pool_mr = rmm::mr::pool_memory_resource<pinned_mr>;
+using arena_mr       = rmm::mr::arena_memory_resource<cuda_mr>;
+using fixed_mr       = rmm::mr::fixed_size_memory_resource<cuda_mr>;
+using binning_mr     = rmm::mr::binning_memory_resource<pool_mr>;
+
+inline std::shared_ptr<mr_factory_base> mr_factory_dispatch(std::string name)
+{
+  if (name == "CUDA") {
+    return std::make_shared<mr_factory<cuda_mr, decltype(make_cuda)>>("CUDA", make_cuda);
+  } else if (name == "Host_Pinned") {
+    return std::make_shared<mr_factory<pinned_mr, decltype(make_host_pinned)>>("Host_Pinned",
+                                                                               make_host_pinned);
+  } else if (name == "CUDA_Async") {
+    return std::make_shared<mr_factory<cuda_async_mr, decltype(make_cuda_async)>>("CUDA_Async",
+                                                                                  make_cuda_async);
+  } else if (name == "Managed") {
+    return std::make_shared<mr_factory<managed_mr, decltype(make_managed)>>("Managed",
+                                                                            make_managed);
+  } else if (name == "Pool") {
+    return std::make_shared<mr_factory<pool_mr, decltype(make_pool)>>("Pool", make_pool);
+  } else if (name == "Host_Pinned_Pool") {
+    return std::make_shared<mr_factory<pinned_pool_mr, decltype(make_host_pinned_pool)>>(
+      "Host_Pinned_Pool", make_host_pinned_pool);
+  } else if (name == "Arena") {
+    return std::make_shared<mr_factory<arena_mr, decltype(make_arena)>>("Arena", make_arena);
+  } else if (name == "Binning") {
+    return std::make_shared<mr_factory<binning_mr, decltype(make_binning)>>("Binning",
+                                                                            make_binning);
+  } else if (name == "Fixed_Size") {
+    return std::make_shared<mr_factory<fixed_mr, decltype(make_fixed_size)>>("Fixed_Size",
+                                                                             make_fixed_size);
+  } else {
+    return std::make_shared<mr_factory<cuda_mr, decltype(make_cuda)>>("Error", make_cuda);
+  }
+}
+
+/// Test fixture class value-parameterized on different `mr_factory`s
+struct mr_ref_test : public ::testing::TestWithParam<std::string> {
+  void SetUp() override
+  {
+    factory_obj = mr_factory_dispatch(GetParam());
+    if (factory_obj->skip_test) {
+      GTEST_SKIP() << "Skipping tests since the memory resource is not supported with this CUDA"
```

Comment:
Needs a trailing space to fit with the next line.
```suggestion
      GTEST_SKIP() << "Skipping tests since the memory resource is not supported with this CUDA "
```